packages/main/src/SliderBase.js

```diff
@@ -0,0 +1,623 @@
+		 * @public
+		 */
+		labelInterval: {
+			type: Integer,
+			defaultValue: 0,
+		},
+		/**
+		 * Enables tick marks visualization for each step. The step value must not be set to 0.
+		 * <br><br>
+		 *
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		 tickmarks: {
+			type: Boolean,
+		},
+		/**
+		 * Enables handle tooltip displaying the current value.
+		 * <br><br>
+		 *
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		showTooltip: {
+			type: Boolean,
+		},
+		/**
+		 * Defines whether the <code>ui5-slider</code> is in disabled state.
+		 * <br><br>
+		 *
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		disabled: {
+			type: Boolean,
+		},
+		/**
+		 * @private
+		 */
+		_tooltipVisibility: {
+			type: String,
+			defaultValue: "hidden",
+		},
+		_labelsOverlapping: {
+			type: Boolean,
+		},
+		_hiddenTickmarks: {
+			type: Boolean,
+		},
+	},
+	slots: /** @lends sap.ui.webcomponents.main.SliderBase.prototype */ {
+		/**
+		 * Defines the text of the <code>ui5-range-slider</code>.
+		 * <br><b>Note:</b> Although this slot accepts HTML Elements, it is strongly recommended that you only use text in order to preserve the intended design.
+		 *
+		 * @type {Node[]}
+		 * @slot
+		 * @public
+		 */
+		"default": {
+			type: Node,
+		},
+	},
+	events: /** @lends sap.ui.webcomponents.main.Slider.prototype */ {
+		/**
+		 * Fired when the value changes and the user has finished interacting with the slider.
+		 *
+		 * @event
+		 * @public
+		*/
+		change: {},
+		/**
+		 * Fired when the value changes due to user interaction that is not yet finished - during mouse/touch dragging.
+		 *
+		 * @event
+		 * @public
+		*/
+		input: {},
+	},
+};
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.SliderBase
+ * @extends sap.ui.webcomponents.base.UI5Element
+ * @tagname ui5-slider
+ * @public
+ */
+class SliderBase extends UI5Element {
+	constructor() {
+		super();
+		this._resizeHandler = this._handleResize.bind(this);
+		this._moveHandler = this._handleMove.bind(this);
+		this._upHandler = this._handleUp.bind(this);
+
+		this.TICKMARK_COLOR_MAP = {
+			sap_fiori_3: "#89919a",
+			sap_fiori_3_dark: "#89919a",
+			sap_fiori_3_hcw: "#000000",
+			sap_fiori_3_hcb: "#ffffff",
+			sap_belize: "#bfbfbf",
+			sap_belize_hcw: "#000000",
+			sap_belize_hcb: "#ffffff",
+		};
+
+		this._stateStorage = {
+			step: null,
+			min: null,
+			max: null,
+		};
+	}
+
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return styles;
+	}
+
+	static get UP_EVENTS() {
+		return ["mouseup", "pointerup", "touchend"];
+	}
+
+	static get MOVE_EVENT_MAP() {
+		return {
+			mousedown: "mousemove",
+			pointerdown: "pointermove",
+			touchstart: "touchmove",
+		};
+	}
+
+	get labelItems() {
+		return this._labelItems;
```

Comment:
_labelItems has to be declared in the constructor (if not in the metadata) and cleared upon onExitDom